zipline/pipeline/data/dataset.py

```diff
@@ -115,14 +118,15 @@ class BoundColumn(LoadableTerm):
     inputs = ()
     window_safe = True
 
-    def __new__(cls, dtype, missing_value, dataset, name):
+    def __new__(cls, dtype, missing_value, dataset, name, ndim):
```

Comment:
Agreed on forwarding `dataset.ndim`. Not sure what my rationale on passing `ndim`all the way through was, but that seems much cleaner